exercises/guillaume_communie/ex02_oo_basics/inc/Shape.h

```diff
@@ -0,0 +1,66 @@
+#ifndef SHAPE_H
+#define SHAPE_H
+
+#include <string>
+
+/**
+ * A shape
+ */
+class Shape {
+public:
+    /**
+     * Create a shape by a given type name and a number of sides
+     * @param[in] t The type name
+     * @param[in] n The number of sides
+     */
+    Shape(std::string t, int n);
+
+    /**
+     * Delete a shape
+     */
+    //~Shape(void);
+
+    /**
+     * Get the type name of the shape
+     * @return The type name
+     */
+    std::string getType(void);
+
+    /**
+     * Get the number of sides
+     * @return The shape number of sides
+     */
+    int getSides(void);
+
+    /**
+     * Get the perimeter
+     * @return The shape perimeter
+     */
+    virtual double getPerimeter(void) = 0;
```

Comment:
For the overrides of the virtuals, you can make them explicit with `override` keyword.